DuckDuckGo/SpecialErrorPage/SpecialErrorPageInterfaces/SpecialErrorPageActionHandler.swift

```diff
@@ -26,11 +27,14 @@ import Foundation
 /// advanced information related to the error.
 protocol SpecialErrorPageActionHandler {
     /// Handles the action of navigating to the site associated with the error page
-    func visitSite()
+    @MainActor
+    func visitSite(url: URL, errorData: SpecialErrorData)
```

Comment:
I’m a bit on the fence with this. 
The idea was to have a generic protocol that all the sub-special error handlers should conform to. 
For `MaliciousSiteProtectionNavigationHandler` I need to pass the URL and SpecialErrorData but I don’t need them for the SSL handler. So I feel I’m violating the Interface Segregation principle. what do you think?